### DISCUSSION TITLE
How do I include the ids of useFieldArray items on the values argument of the onSubmit function?

### DISCUSSION THREAD

**Discussion by canrozanes** [react-hook-form/react-hook-form#5426] [Q&A] [answered]
Hey,

I have a form that allows users to create or update existing contact-infos for clients. 

As you can see below, Person to PhoneNumber is a one-to-many relationship so I figured it'd be a good idea to use the useFieldArray hook.

```
export interface Person {
  id: number;
  name: string;
  phoneNumbers: PhoneNumber[];
}

export interface PhoneNumber {
  id: number
  number: string
  description: string
}
```

On the `/update` route of the application, I need to keep track of all the PhoneNumbers that are unchanged, created, updated and/or deleted so I can run the correct operations in the backend accordingly.

To do that, I need to submit the id of the PhoneNumber back to the back-end. So that:
If the PhoneNumber.id does not exists in the backend, I run a create operation.
If the PhoneNumber.id exists on the backend, I run an update operation.

However, when I currently submit the form, the form values does not include PhoneNumber.id field.

Is there a way to enable submitting the id's of the FieldArray items?

Thanks

**Comment by bluebill1049** [comment, marked as answer]
Please provide ur own Id and u can register the value as hidden input.

**Reply by canrozanes** [reply]
Thanks for your prompt response! 

What is a hidden input? Is it literally a controlled input that has css property { display: none } or is there a cleaner API i can use to set this up?

**Reply by JamesWong1999** [reply]
`<input {...register('id")} type="hidden" defaultValue={yourId} />` 👍

**Reply by canrozanes** [reply]
ah. that makes sense thanks.
